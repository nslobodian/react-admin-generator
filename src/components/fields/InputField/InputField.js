import React from 'react'
import s from './InputField.scss'

export const InputField = ({ input, meta: { submitFailed, error, asyncValidating, touched }, forceShowErrors, textarea, className, label, noBlur, ...props }) => (
  <div className={asyncValidating ? 'relative' : className || ''}>
    { label && <div className='text-muted'>{label}</div>}
    {textarea
      ? <textarea {...input}
        className='form-control'
        {...props}
        onBlur={noBlur ? () => {} : input.onBlur}
      />
      : <input
        {...input}
        className='form-control'
        type='text'
        onBlur={noBlur ? () => {} : input.onBlur}
        {...props}
      />
    }
    {asyncValidating && <div className={s.asyncValidating} />}

    {(submitFailed || forceShowErrors || touched) && error &&
    <div className='alert alert-danger m-t-1' role='alert'>{error.err || error}</div>}
  </div>
)

export default InputField
