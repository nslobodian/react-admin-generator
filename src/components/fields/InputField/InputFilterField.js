import React from 'react'
import compose from 'recompose/compose'
import withState from 'recompose/withState'
import withHandlers from 'recompose/withHandlers'

export const enhance = compose(
  withState('debounce', 'setDebounce', null),
  withHandlers({
    handleChange: ({ setDebounce, input: { onChange, name }, changeAttempt, debounce }) => ({ target: { value } }) => {
      onChange(value)
      clearTimeout(debounce)
      setDebounce(setTimeout(() => {
        changeAttempt(name, value)
      }, 500))
    },
  })
)

export default enhance(({ input: { value, name, onBlur }, placeholder = '', handleChange, className, type = 'text' }) =>
  <input value={value} name={name} onChange={handleChange} placeholder={placeholder} onBlur={onBlur} className={className} type={type} />
)
