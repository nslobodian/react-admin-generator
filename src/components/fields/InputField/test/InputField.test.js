import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { InputField } from '../InputField'

describe('(Components) InputField', () => {
  it('Should render component properly', () => {
    const props = {
      input: {
        value: 1,
        onChange: jest.fn(),
      },
      meta: {
        asyncValidating: jest.fn(),
        submitFailed: 'submitFailed',
        touched: 'touched',
      },
      forceShowErrors: jest.fn(),
      textarea: true,
      className: 'className',
      label: 'label',
    }
    const wrapper = shallow(
      <InputField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render component properly without textarea', () => {
    const props = {
      input: {
        value: 1,
        onChange: jest.fn(),
      },
      meta: {
        asyncValidating: jest.fn(),
        touched: 'touched',
      },
      forceShowErrors: jest.fn(),
      textarea: false,
      label: 'label',
    }
    const wrapper = shallow(
      <InputField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('Should render component properly testing with error', () => {
    const props = {
      input: {
        value: 1,
        onChange: jest.fn(),
      },
      meta: {
        touched: 'touched',
        error: 'error',
      },
      textarea: false,
      label: 'label',
    }
    const wrapper = shallow(
      <InputField
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
