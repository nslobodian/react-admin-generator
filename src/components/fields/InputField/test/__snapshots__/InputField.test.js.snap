// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Components) InputField Should render component properly 1`] = `
<div
  className="relative"
>
  <div
    className="text-muted"
  >
    label
  </div>
  <textarea
    className="form-control"
    onChange={[Function]}
    value={1}
  />
  <div
    className="asyncValidating"
  />
</div>
`;

exports[`(Components) InputField Should render component properly testing with error 1`] = `
<div
  className=""
>
  <div
    className="text-muted"
  >
    label
  </div>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value={1}
  />
  <div
    className="alert alert-danger m-t-1"
    role="alert"
  >
    error
  </div>
</div>
`;

exports[`(Components) InputField Should render component properly without textarea 1`] = `
<div
  className="relative"
>
  <div
    className="text-muted"
  >
    label
  </div>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value={1}
  />
  <div
    className="asyncValidating"
  />
</div>
`;
